File: bun/src/repositories/product-repository.ts
import type { Product } from '../models/product.js';

export class ProductRepository {
  private products: Product[];

  constructor() {
    this.products = [];
    for (let i = 1; i <= 10000; i++) {
      const userId = ((i % 5) + 1).toString();
      const categoryId = ((i % 5) + 1).toString();
      this.products.push({
        id: i.toString(),
        name: `Product ${i}`,
        description: `Description for product ${i}`,
        userId,
        price: i * 1.5,
        categoryId,
      });
    }
  }

  getAllProducts(): Product[] {
    return this.products;
  }

  getProductById(id: string): Product | undefined {
    return this.products.find(p => p.id === id);
  }
}

File: aspnetcore/Repositories/ProductRepository.cs
using local_ly_dotnet.Models;

namespace local_ly_dotnet.Repositories;

public class ProductRepository
{
    private readonly List<Product> _products;

    public ProductRepository()
    {
        _products = [];
        for (int i = 1; i <= 10_000; i++)
        {
            _products.Add(
                new Product
                {
                    Id = i.ToString(),
                    Name = $"Product {i}",
                    Description = $"Description for product {i}",
                    UserId = ((i % 5) + 1).ToString(),
                    Price = i * 1.5m,
                    CategoryId = ((i % 5) + 1).ToString(),
                }
            );
        }
    }

    public List<Product> GetAllProducts() => _products;

    public Product? GetProductById(string id) => _products.FirstOrDefault(p => p.Id == id);
}
